### PR TITLE
Add MinValue/MaxValue helpers on our NumericAttributeTraits.

### LINKED ISSUE #35328: [Feature]  NumericAttributeTraits max/min
### Feature description

NumericAttributeTraits should have a method allowing to obtain the min/max value for the specific type associatgeed with it.

### Platform

all

### Platform Version(s)

_No response_

### Anything else?

This was raised here:
https://github.com/project-chip/connectedhomeip/pull/32632#discussion_r1734866016

This would simplify steps to verify validity of EFS for the Scenes Management cluster.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
Uses the new helpers in SceneHandlerImpl.

Fixes https://github.com/project-chip/connectedhomeip/issues/35328

Review note: git diff is a bit annoyingly confused on the odd-sized-integers bits.  The PR does not in fact change anything about Encode() or ToAttributeStoreRepresentation().

### CODE REVIEW

File: src/app/util/attribute-storage-null-handling.h [hunk 1]
```diff
@@ -127,6 +127,39 @@ struct NumericAttributeTraits
...
+    static WorkingType MinValue(bool isNullable)
+    {
+        if constexpr (!std::is_signed_v<WorkingType>)
+        {
+            return 0;
+        }
+
+        if (isNullable)
+        {
+            // Smallest negative value is excluded for nullable types.
```
Comment (andy31415) [inline src/app/util/attribute-storage-null-handling.h:141]:
```suggestion
            // Smallest negative value is excluded for nullable UNSIGNED types.
```

File: src/app/util/attribute-storage-null-handling.h [hunk 2]
```diff
@@ -127,6 +127,39 @@ struct NumericAttributeTraits
...
+    static WorkingType MaxValue(bool isNullable)
+    {
+        if constexpr (std::is_signed_v<WorkingType>)
+        {
+            return std::numeric_limits<WorkingType>::max();
+        }
+
+        if (isNullable)
+        {
+            // Largest value is excluded for nullable types.
```
Comment (andy31415) [inline src/app/util/attribute-storage-null-handling.h:157]:
```suggestion
            // Largest value is excluded for nullable SIGNED types.
```

**Review by lpbeliveau-silabs — APPROVED** [review]

File: src/app/util/attribute-storage-null-handling.h (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/util/attribute-storage-null-handling.h:141]: You mean signed, right?

File: src/app/util/attribute-storage-null-handling.h (hunk 2 above)
Comment (bzbarsky-apple) [reply src/app/util/attribute-storage-null-handling.h:157]: You mean unsigned, right?

File: src/app/util/odd-sized-integers.h [hunk 3]
```diff
@@ -200,35 +200,52 @@ struct NumericAttributeTraits<OddSizedInteger<ByteSize, IsSigned>, IsBigEndian>
...
-    }
+        // Since WorkingType has at least one extra byte, none of our bitshifts
+        // overflow.
+        if constexpr (IsSigned)
+        {
+            return (static_cast<WorkingType>(1) << (8 * ByteSize - 1)) - 1;
+        }
 
-    static uint8_t * ToAttributeStoreRepresentation(StorageType & value) { return value; }
+        constexpr WorkingType unsignedMax = (static_cast<WorkingType>(1) << (8 * ByteSize));
```
[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply src/app/util/odd-sized-integers.h]: Nice catch!  Also, good thing we have tests.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**